src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -258,14 +258,33 @@ void IdealLoopTree::compute_profile_trip_cnt(PhaseIdealLoop *phase) {
 // Return nonzero index of invariant operand for an associative
 // binary operation of (nonconstant) invariant and variant values.
 // Helper for reassociate_invariants.
-int IdealLoopTree::find_invariant(Node* n, PhaseIdealLoop *phase) {
+int IdealLoopTree::find_invariant(Node* n, PhaseIdealLoop* phase) {
   bool in1_invar = this->is_invariant(n->in(1));
   bool in2_invar = this->is_invariant(n->in(2));
   if (in1_invar && !in2_invar) return 1;
   if (!in1_invar && in2_invar) return 2;
   return 0;
 }
 
+//---------------------is_associative_cmp-------------------------
+// Return TRUE if "n" is an associative cmp node. A cmp node is
+// associative if it is only used for equals or not-equals
+// comparisons of integers or longs. We cannot reassociate
+// non-equality comparisons due to possibility of overflow.
+bool IdealLoopTree::is_associative_cmp(Node* n) {
+  if (n->Opcode() != Op_CmpI && n->Opcode() != Op_CmpL) {
+    return false;
+  }
+  for (DUIterator i = n->outs(); n->has_out(i); i++) {
+    BoolNode* boolOut = n->out(i)->isa_Bool();
+    if (boolOut == nullptr || !(boolOut->_test._test == BoolTest::eq ||
+                                boolOut->_test._test == BoolTest::ne)) {
```

Comment:
We should use underscores instead of camelCase.
```suggestion
    BoolNode* bool_out = n->out(i)->isa_Bool();
    if (bool_out == nullptr || !(bool_out->_test._test == BoolTest::eq ||
                                 bool_out->_test._test == BoolTest::ne)) {
```

Reply:
Agree, this is the C++ style. CamelCase is Java style.